src/host/manual.rs

```diff
@@ -0,0 +1,36 @@
+use crate::bus::Bus;
+use crate::host::{Host, HostConfig};
+use crate::MacAddress;
+use embedded_nal::Ipv4Addr;
+
+pub struct Manual {
+    is_setup: bool,
+    settings: HostConfig,
+    current: HostConfig,
```

Comment:
What do you mean?

Reply:
The [`Device<Manual>`](https://github.com/kellerkindt/w5500/blob/master/src/device.rs#L10) is about 38 bytes large in memory:

 * ~0 bytes for the SPI-Bus (depending on the optimization)
 *  37 bytes for HostImpl (for [`Manual`](https://github.com/kellerkindt/w5500/blob/master/src/host/manual.rs#L6))
    * 1 byte for `is_setup` (we could eliminate this to a generic type, e.g. `Manual<Setup>` and `Manual<NotSetup>`)
    * 36 bytes because of 2x [`HostConfig`](https://github.com/kellerkindt/w5500/blob/master/src/host/mod.rs#L11)
      * 6 bytes for `mac`
      * 4 bytes for `ip`
      * 4 bytes for `gateway`
      * 4 bytes for `subnet`
* 1 byte for the sockets

What I mean is, that "just for" the feature of not having  to always send  the whole config in [`Host::refresh`](https://github.com/kellerkindt/w5500/blob/master/src/host/mod.rs#L31) all `Device` instances consume 18 additional bytes in memory. Assuming that everyone is okay with spending 18 bytes for having the current configuration cached in the first place. In my personal projects I neither need the current state nor the "shall be" state, which means the `Device` instance is - for me - 36/37 bytes larger than needed.

---

What I just noticed, If we were to pass the "shall value" `HostConfig` into  [`Host::refresh`](https://github.com/kellerkindt/w5500/blob/master/src/host/mod.rs#L31) we could free up 18 bytes. Because of the trait we would probably require a associated type? Like:

```rust
pub trait Host {
    type Config;

    /// Gets (if necessary) and sets the host settings on the chip
    fn refresh<SpiBus: Bus>(&mut self, bus: &mut SpiBus, new_config: &Self::Config) -> Result<(), SpiBus::Error>;
//...
}
```

Reply:
You raise some good points.  I'll work on cutting this down.